core/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -97,6 +97,24 @@ public static boolean isIngestNode(Settings settings) {
     private final Version version;
     private final Set<Role> roles;
 
+
+    /**
+     * Creates a new {@link DiscoveryNode}
+     * <p>
+     * <b>Note:</b> if the version of the node is unknown {@link Version#minimumCompatibilityVersion()} should be used for the current
+     * version. it corresponds to the minimum version this elasticsearch version can communicate with. If a higher version is used
+     * the node might not be able to communicate with the remove node. After initial handshakes node versions will be discovered
+     * and updated.
+     * </p>
+     *
+     * @param id               the nodes unique (persistent) node id. This constructor will auto generate a random ephemeral id.
+     * @param address          the nodes transport address
+     * @param version          the version of the node
+     */
+    public DiscoveryNode(final String id, TransportAddress address, Version version) {
```

Comment:
can you remove this overloaded constructor? We already have too many of them. AFAICS this new one is only used in one place (in a test), where we can easily just provide the 2 extra arguments.